src/cephadm/cephadm

```diff
@@ -4011,6 +4040,10 @@ def command_adopt():
                                   legacy_dir=args.legacy_dir)
     if not fsid:
         raise Error('could not detect legacy fsid; set fsid in ceph.conf')
+
+    if not args.skip_pull:
+        _pull_image(args.image, fsid)
+
```

Comment:
that's exactly the problem: we can no longer pull the image, without first deploying the cluster.